docs/core-features/table_lookups_basics.md

```diff
@@ -0,0 +1,186 @@
+
+## Direct TLU
+
+Direct TLU stands for instructions of the form `y = T[i]`, for some table `T` and some index `i`. One can use the `fhe.LookupTable` to define the table values, and then use it on scalars or tensors.
+
+```python
+from concrete import fhe
+
+table = fhe.LookupTable([2, -1, 3, 0])
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return table[x]
+
+inputset = range(4)
+circuit = f.compile(inputset)
+
+assert circuit.encrypt_run_decrypt(0) == table[0] == 2
+assert circuit.encrypt_run_decrypt(1) == table[1] == -1
+assert circuit.encrypt_run_decrypt(2) == table[2] == 3
+assert circuit.encrypt_run_decrypt(3) == table[3] == 0
+```
+
+```python
+from concrete import fhe
+import numpy as np
+
+table = fhe.LookupTable([2, -1, 3, 0])
+
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return table[x]
+
+
+inputset = [np.random.randint(0, 4, size=(2, 3)) for _ in range(10)]
+circuit = f.compile(inputset)
+
+sample = [
+    [0, 1, 3],
+    [2, 3, 1],
+]
+expected_output = [
+    [2, -1, 0],
+    [3, 0, -1],
+]
+actual_output = circuit.encrypt_run_decrypt(np.array(sample))
+
+assert np.array_equal(actual_output, expected_output)
+```
+
+`LookupTable` mimics array indexing in Python, which means if the lookup variable is negative, the table is looked up from the back.
+
+## Multi TLU
+
+Multi TLU stands for instructions of the form `y = T[j][i]`, for some set of tables `T`, some table-index `j` and some index `i`. One can use the `fhe.LookupTable` to define the table values, and then use it on scalars or tensors.
+
+```python
+from concrete import fhe
+import numpy as np
+
+squared = fhe.LookupTable([i ** 2 for i in range(4)])
+cubed = fhe.LookupTable([i ** 3 for i in range(4)])
+
+table = fhe.LookupTable([
+    [squared, cubed],
+    [squared, cubed],
+    [squared, cubed],
+])
+
+@fhe.compiler({"x": "encrypted"})
+def f(x):
+    return table[x]
+
+inputset = [np.random.randint(0, 4, size=(3, 2)) for _ in range(10)]
+circuit = f.compile(inputset)
+
+sample = [
+    [0, 1],
+    [2, 3],
+    [3, 0],
+]
+expected_output = [
+    [0, 1],
+    [4, 27],
+    [9, 0]
+]
+actual_output = circuit.encrypt_run_decrypt(np.array(sample))
+
+assert np.array_equal(actual_output, expected_output)
+```
+
+### TLU on the most significant bits
+
+For lot of use-cases, like for example in Machine Learning, it is possible to replace the table lookup `y = T[i]` by some `y = T'[i']`, where `i'` only has the most significant bits of `i` and `T'` is a much shorter table, and still maintain a good accuracy of the function. The interest of such a method stands in the fact that, since the table `T'` is much smaller, the corresponding TLU will be done much more quickly.
```

Comment:
Here as an user I don't understand why I need to do that, we should explain at least in a on sentence that the input bitwidth of TLU is one of the most important optimization to be done?

Reply:
ok yes

Reply:
done, you'll tell me if you like the new part